code/modules/components/ai/atmos.dm

```diff
@@ -0,0 +1,94 @@
+/datum/component/ai/atmos_checker
+	//Atmos effect - Yes, you can make creatures that require plasma or co2 to survive. N2O is a trace gas and handled separately, hence why it isn't here. It'd be hard to add it. Hard and me don't mix (Yes, yes make all the dick jokes you want with that.) - Errorage
+	var/min_oxy = 5
+	var/max_oxy = 0					//Leaving something at 0 means it's off - has no maximum
+	var/min_tox = 0
+	var/max_tox = 1
+	var/min_co2 = 0
+	var/max_co2 = 5
+	var/min_n2 = 0
+	var/max_n2 = 0
+	var/unsuitable_damage = 2	//This damage is taken when atmos doesn't fit all the requirements above
+
+	var/minbodytemp = 250
+	var/maxbodytemp = 350
+	var/heat_damage_per_tick = 3	//amount of damage applied if animal's body temperature is higher than maxbodytemp
+	var/cold_damage_per_tick = 2	//same as heat_damage_per_tick, only if the bodytemperature it's lower than minbodytemp
+	var/fire_alert = 0
+	var/oxygen_alert = 0
+	var/toxins_alert = 0
+
+/datum/component/ai/atmos_checker/RecieveSignal(var/message_type, var/list/args)
+	switch(message_type)
+		if("life")
+			OnLife()
+		else
+			..(message_type, args)
+
+/datum/component/ai/atmos_checker/proc/OnLife()
+	if(isliving(container.holder))
+		if(container.holder & INVULNERABLE)
+			return 1
+
+		var/atmos_suitable = 1
+
+		var/atom/A = container.holder.loc
+
+		if(isturf(A))
+			var/turf/T = A
+			var/datum/gas_mixture/Environment = T.return_air()
+
+			if(Environment)
+				if(abs(Environment.temperature - controller.getBodyTemperature()) > 40)
```

Comment:
Considering this component will be a bit more general I'd say making this a variable should be a good idea.